controllers/utils/dns.go

```diff
@@ -47,7 +47,12 @@ func (l DNSList) String() string {
 
 // Dig returns a list of IP addresses for a given FQDN by using the dns servers from edgeDNSServers
 // dns servers are tried one by one from the edgeDNSServers and if there is a non-error response it is returned and the rest is not tried
-func Dig(fqdn string, edgeDNSServers ...DNSServer) (ips []string, err error) {
+func Dig(fqdn string, maxRecursion int, edgeDNSServers ...DNSServer) (ips []string, err error) {
+	if maxRecursion < 1 {
```

Comment:
Nitpick:

I think we should only fail if maxRecursion < 0.
If the domain name can be resolved with a single query there is no recursion and it is successful.

An alternative is to rename the variable to maxQueries. Like that we can keep the < 1